### PR TITLE
bug setActiveConnection always error

### PR DESCRIPTION
**Pull request by wanyvic**

### CODE REVIEW

**Comment by MariusVanDerWijden** [comment]
Isn't this equal to this? Would save us the found bool
```
    for (size_t idx = 0; idx < m_Settings.connections.size(); idx++)
        if (boost::iequals(m_Settings.connections[idx]->str(), _connstring))
        {
            setActiveConnectionCommon(idx);
            return;
        }
    throw std::runtime_error("Not found.");
```

**Comment by wanyvic** [comment]
> Isn't this equal to this? Would save us the found bool
> 
> ```
>     for (size_t idx = 0; idx < m_Settings.connections.size(); idx++)
>         if (boost::iequals(m_Settings.connections[idx]->str(), _connstring))
>         {
>             setActiveConnectionCommon(idx);
>             return;
>         }
>     throw std::runtime_error("Not found.");
> ```
> 
> Cool find by the way, Don't know how this could slide by

much better

File: libpoolprotocols/PoolManager.cpp [hunk 1]
```diff
@@ -334,7 +334,7 @@ void PoolManager::setActiveConnection(std::string& _connstring)
         if (boost::iequals(m_Settings.connections[idx]->str(), _connstring))
         {
             setActiveConnectionCommon(idx);
-            break;
+            return;
         }
     if (!found)
```
Comment (MariusVanDerWijden) [inline libpoolprotocols/PoolManager.cpp]: can you remove this?

Comment (MariusVanDerWijden) [reply libpoolprotocols/PoolManager.cpp]: and the declaration of found

Comment (wanyvic) [reply libpoolprotocols/PoolManager.cpp]: > and the declaration of found

done

**Pull request merged**